# Configuration Settings for C94-M8P device
ublox_gps_node:
  ros__parameters:
    debug: 1                    # Range 0-4 (0 means no debug statements will print)

    save:
      mask: 3103                # Save I/O, Message, INF Message, Nav, Receiver
                                # Manager, Antenna, and Logging Configuration
      device: 4                 # Save to EEPROM

    device: /dev/ttyACM0
    frame_id: gps
    rate: 1.0                   # in Hz
    nav_rate: 1                 # [# of measurement cycles], recommended 1 Hz, may
                                # be either 5 Hz (Dual constellation) or
                                # 8 Hz (GPS only)
    dynamic_model: portable     # Airborne < 2G, 2D fix not supported (3D only),
                                # Max Alt: 50km
                                # Max Horizontal Velocity: 250 m/s,
                                # Max Vertical Velocity: 100 m/s
    fix_mode: auto
    enable_ppp: false           # Not supported by C94-M8P
    dr_limit: 0

    # TMODE3 Config
    tmode3: 0                   # Disabled
    sv_in:
      reset: false              # True: disables and re-enables survey-in (resets)
                                # False: Disables survey-in only if TMODE3 is
                                # disabled
      min_dur: 300              # Survey-In Minimum Duration [s]
      acc_lim: 3.0              # Survey-In Accuracy Limit [m]

    uart1:
      baudrate: 19200           # C94-M8P specific
      in: 32                    # RTCM 3
      out: 0                    # No UART out for rover

    gnss:
      glonass: true             # Supported by C94-M8P
      beidou: false             # Supported by C94-M8P
      qzss: true                # Supported by C94-M8P

    dgnss_mode: 3               # Fixed mode

    inf:
      all: true                 # Whether to display all INF messages in console

    # Enable u-blox message publishers
    publish:
      all: true
      aid:
        hui: false

      nav:
        posecef: false
